dds/DCPS/XTypes/TypeObject.h

```diff
@@ -828,6 +828,7 @@ namespace XTypes {
   // ID of a type member
   typedef ACE_CDR::ULong MemberId;
   const ACE_CDR::ULong MEMBER_ID_INVALID = UINT32_MAX;
+  const ACE_CDR::ULong DISCRIMINATOR_ID = 4294967294U;
```

Comment:
```suggestion
  const ACE_CDR::ULong DISCRIMINATOR_ID = UINT32_MAX - 1;
```

Also maybe add a note that this one is implementation specific.